src/WebJobs.Extensions.DurableTask/EndToEndTraceHelper.cs

```diff
@@ -284,6 +284,56 @@ public void ExtensionWarningEvent(string hubName, string functionName, string in
                 LocalSlotName, ExtensionVersion, this.sequenceNumber++);
         }
 
+        public void FunctionSuspended(
+            string hubName,
+            string functionName,
+            string instanceId,
+            string reason)
+        {
+            FunctionType functionType = FunctionType.Orchestrator;
+
+            EtwEventSource.Instance.FunctionSuspended(
+                hubName,
+                LocalAppName,
+                LocalSlotName,
+                functionName,
+                instanceId,
+                reason,
+                functionType.ToString(),
+                ExtensionVersion,
+                IsReplay: false);
+
+            this.logger.LogWarning(
+                "{instanceId}: Function '{functionName} ({functionType})' was suspended. Reason: {reason}. State: {state}. HubName: {hubName}. AppName: {appName}. SlotName: {slotName}. ExtensionVersion: {extensionVersion}. SequenceNumber: {sequenceNumber}.",
+                instanceId, functionName, functionType, reason, FunctionState.Suspended, hubName, LocalAppName,
+                LocalSlotName, ExtensionVersion, this.sequenceNumber++);
+        }
+
+        public void FunctionResumed(
+            string hubName,
+            string functionName,
+            string instanceId,
+            string reason)
+        {
+            FunctionType functionType = FunctionType.Orchestrator;
+
+            EtwEventSource.Instance.FunctionResumed(
+                hubName,
+                LocalAppName,
+                LocalSlotName,
+                functionName,
+                instanceId,
+                reason,
+                functionType.ToString(),
+                ExtensionVersion,
+                IsReplay: false);
+
+            this.logger.LogWarning(
+                "{instanceId}: Function '{functionName} ({functionType})' was resumed. Reason: {reason}. State: {state}. HubName: {hubName}. AppName: {appName}. SlotName: {slotName}. ExtensionVersion: {extensionVersion}. SequenceNumber: {sequenceNumber}.",
+                instanceId, functionName, functionType, reason, FunctionState.Scheduled, hubName, LocalAppName,
```

Comment:
I wanted to double check whether Scheduled would be the right function state?

Reply:
Considering that we didn't introduce a `Resumed` State, like we did for `Suspended`, then I think `Scheduled` is perfectly valid in here.